lib/linter/linter.js

```diff
@@ -1022,7 +1036,7 @@ function runRules(
                 Object.create(sharedTraversalContext),
                 {
                     id: ruleId,
-                    options: getRuleOptions(configuredRules[ruleId]),
+                    options: getRuleOptions(configuredRules[ruleId], applyDefaultOptions && rule.meta?.defaultOptions),
```

Comment:
```suggestion
                    options: getRuleOptions(configuredRules[ruleId], applyDefaultOptions ? rule.meta?.defaultOptions : void 0),
```

It's a bit difficult to follow what happens when `applyDefaultOptions` is `false` and then this passes `false` to `getRuleOptions`.